gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/TicketServiceImpl.java

```
@@ -59,70 +78,138 @@
 
-        if (ticketEntity.getApplication() != null && !ticketEntity.getApplication().isEmpty()) {
-            applicationEntity = applicationService.findById(ticketEntity.getApplication());
-            parameters.put("application", applicationEntity);
-        } else {
-            applicationEntity = null;
+
+            LOGGER.debug("search tickets - Done with {} elements", entities.size());
+
+            return new Page<>(entities,
+                    tickets.getPageNumber() + 1,
+                    (int) tickets.getPageElements(),
+                    tickets.getTotalElements()
+            );
+        } catch (TechnicalException ex) {
+            LOGGER.error("An error occurs while trying to search tickets", ex);
+            throw new TechnicalManagementException("An error occurs while trying to search tickets", ex);
         }
+    }
 
-        parameters.put("content", ticketEntity.getContent().replaceAll("(\r\n|\n)", "<br />"));
-        parameters.put("ticketSubject", ticketEntity.getSubject());
-        final String fromName = user.getFirstname() == null ? user.getEmail() : user.getFirstname() + ' ' + user.getLastname();
-        emailService.sendEmailNotification(
-                new EmailNotificationBuilder()
-                        .replyTo(user.getEmail())
-                        .fromName(fromName)
-                        .to(emailTo)
-                        .copyToSender(ticketEntity.isCopyToSender())
-                        .template(TEMPLATES_FOR_ACTION_SUPPORT_TICKET)
-                        .params(parameters)
-                        .build());
-        sendUserNotification(user, api, applicationEntity);
+    @Override
+    public TicketEntity findById(String ticketId) {
+        try {
+            return ticketRepository
+                    .findById(ticketId)
+                    .map(this::getApiNameAndApplicationName)
+                    .map(this::convert)
+                    .orElse(null);
```

should throw an exception instead

---

resolved